test/integration/com/lucastex/grails/fileuploader/FileUploaderServiceSpec.groovy

```diff
@@ -0,0 +1,212 @@
+        File testFile = getTestFile()
+
+        when:
+        try {
+            ufileInstance = fileUploaderService.saveFile(group, testFile)
+        } catch (Exception e) {
+            configException = e
+        }
+
+        then: "Method should throw exception for missing file uploader plugin configuration."
+        assert configException.message == "No config defined for group [${group}]. " +
+            "Please define one in your Config file."
+
+        cleanup: "Deleting test file for other test cases."
+        testFile?.delete()
+    }
+
+    void "Test: saveFile() method should throw exception for invalid file extention."() {
+        given: "Applying FileUploader configuration and creating file with unauthorized extension"
+        setupConfig()
+        FileUploaderServiceException unauthorizedExtension = null
+        File testFile = getTestFile("test-logo.tif")
+
+        when:
+        try {
+            ufileInstance = fileUploaderService.saveFile(group, testFile)
+        } catch (Exception e) {
+            unauthorizedExtension = e
+        }
+
+        then: "Method should throw exception for invalid file extention."
+        String exceptionMessage = "The file you sent has an unauthorized extension (tif)." +
+            " Allowed extensions for this upload are " + 
+            "${grailsApplication.config.fileuploader.logo.allowedExtensions}"
+
+        assert unauthorizedExtension?.message == exceptionMessage
+
+        cleanup: "Deleting test file for other test cases."
+        testFile?.delete()
+    }
+
+    void "Test: saveFile() method should throw exception for Max file size."() {
+        given: "Applying FileUploader configuration and updating file size configuration."
+        setupConfig()
+        grailsApplication.config.fileuploader.logo.maxSize = 1024 // 1kb
+        FileUploaderServiceException maxFileSizeExtension = null
+        File testFile = getTestFile()
+
+        when:
+        try {
+            ufileInstance = fileUploaderService.saveFile(group, testFile)
+        } catch (Exception e) {
+            maxFileSizeExtension = e
+        }
+
+        then: "Method should throw exception for Max file size."
+        String exceptionMessage = "Sent file is bigger than allowed. Max file size is 1 kb"
+        assert maxFileSizeExtension?.message == exceptionMessage
+
+        cleanup: "Deleting test file for other test cases."
+        testFile?.delete()
+    }
+
+    void "Test: saveFile() method should save file successfully."() {
+        given: "Applying FileUploader configuration."
+        setupConfig()
+        FileUploaderServiceException exception = null
+        File testFile = getTestFile()
+
+        when:
+        try {
+            ufileInstance = fileUploaderService.saveFile(group, testFile)
+        } catch (Exception e) {
+            exception = e
+        }
+
+        then: "Method should create UFile Instance successfully."
```

Comment:
**method should create UFile instance and file should me moved to configured location locally**